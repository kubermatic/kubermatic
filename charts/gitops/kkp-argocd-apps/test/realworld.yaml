# Copyright 2024 The Kubermatic Kubernetes Platform contributors.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

environment: dev
seed: demo-master
isMasterSeed: true

kkpBasePath: "."

repoURL: https://github.com/dharapvj/kkp-using-argocd.git
valuesPathPrefix: "." # This is prefix wrt environment folder e.g. /dev/

# KKP Core
nginx:
  enable: true
certManager:
  enable: true
oauth:
  enable: true
  autoSync: true
seedSettings:
  enable: true
  autoSync: true
seedExtras:
  enable: true
  autoSync: true

#monitoring
prometheus:
  enable: true
  autoSync: true
alertManager:
  enable: true
  autoSync: true
grafana:
  enable: true
  autoSync: true
kubeStateMetrics:
  enable: true
  autoSync: true
nodeExporter:
  enable: true
  autoSync: true
blackboxExporter:
  enable: true
  autoSync: true
# FIXME: Get thanos deployment to work as we expect it
thanos:    # This gets installed from upstream (bitnami / banzai / stevehipwell) directly
  enable: false
  autoSync: true
iapSeedMla:
  enable: true
  autoSync: true

#logging
promtail:
  enable: true
  autoSync: true
loki:
  enable: true
  autoSync: true

#backup
velero:
  enable: true
  autoSync: true

# storage
minio:
  enable: true
  autoSync: true

#user cluster mla
userMla:
  enable: true # master switch to control whether you want to deploy user-cluster MLA or not
  minio:
    enable: true
    autoSync: true
  minioLifecycleMgr:
    enable: true
    autoSync: true
  grafana:
    enable: true
    autoSync: true
  consul:
    enable: true
    autoSync: true
  cortex:
    enable: true
    autoSync: true
  lokiDistributed:
    enable: true
    autoSync: true
  alertmanagerProxy:
    enable: true
    autoSync: true
  iapMla:
    enable: true
    autoSync: true
  secrets:
    enable: true
    autoSync: true
