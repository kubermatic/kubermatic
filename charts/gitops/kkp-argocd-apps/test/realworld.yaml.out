---
# Source: argocd-apps/templates/argocd-apps-storage.yaml
apiVersion: v1
kind: Namespace
metadata:
  name: minio
---
# Source: argocd-apps/templates/argocd-apps-usermla.yaml
# in case of mla, we define a configmap so - we will need to create NS explicitly.
apiVersion: v1
kind: Namespace
metadata:
  name: mla
---
# Source: argocd-apps/templates/argocd-apps-storage.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: minio-bucket-user-create-script
  namespace: minio
data:
  minio-create.sh: |
    #! /bin/bash
    set -uo pipefail
    # set -x
    max_retry=60 # wait for 1 hour for minio to become available
    counter=0
    until :> /dev/tcp/minio.minio.svc.cluster.local/9000
    do
      sleep 60
      [[ counter -eq $max_retry ]] && echo "Failed! Maximum retries reached!" && exit 2
      ((counter++))
      echo "Minio not available. Trying again. Try #$counter"
    done
    set -e
    mc config host add minio http://minio.minio.svc.cluster.local:9000 "$MINIO_ACCESS_KEY" "$MINIO_SECRET_KEY"
    mc mb --ignore-existing minio/usercluster-etcd-backups
    mc mb --ignore-existing minio/metering-demo-master-seed
    # mc admin user add minio XX YY
    # mc admin policy attach minio readwrite --user XX

    mc mb --ignore-existing minio/velero-backup-dev
    # mc admin user add minio AA BB
    # mc admin policy attach minio readwrite --user AA
---
# Source: argocd-apps/templates/argocd-apps-usermla.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: cortex-runtime-config
  namespace: mla
data:
  runtime-config.yaml: 'overrides:'
---
# Source: argocd-apps/templates/argocd-apps-storage.yaml
apiVersion: batch/v1
kind: Job
metadata:
  name: create-minio-buckets-users
  namespace: minio
spec:
  backoffLimit: 2
  template:
    spec:
      containers:
        - name: mc
          image: quay.io/minio/mc
          command:
            - /bin/sh
            - -c
            - /script/minio-create.sh
          volumeMounts:
            - name: minio-script
              mountPath: "/script"
          env:
            - name: MINIO_ACCESS_KEY
              valueFrom:
                secretKeyRef:
                  name: minio
                  key: accessKey
            - name: MINIO_SECRET_KEY
              valueFrom:
                secretKeyRef:
                  name: minio
                  key: secretKey
      volumes:
        - name: minio-script
          configMap:
            name: minio-bucket-user-create-script
            defaultMode: 0500
      restartPolicy: Never
---
# Source: argocd-apps/templates/argocd-apps-backup.yaml
# Copyright 2024 The Kubermatic Kubernetes Platform contributors.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
---
# Source: argocd-apps/templates/argocd-apps-kkp-core.yaml
# Copyright 2024 The Kubermatic Kubernetes Platform contributors.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
---
# Source: argocd-apps/templates/argocd-apps-logging.yaml
# Copyright 2024 The Kubermatic Kubernetes Platform contributors.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
---
# Source: argocd-apps/templates/argocd-apps-monitoring.yaml
# Copyright 2024 The Kubermatic Kubernetes Platform contributors.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
---
# Source: argocd-apps/templates/argocd-apps-storage.yaml
# Copyright 2024 The Kubermatic Kubernetes Platform contributors.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
---
# Source: argocd-apps/templates/argocd-apps-storage.yaml
# create required buckets as needed
---
# Source: argocd-apps/templates/argocd-apps-usermla.yaml
# Copyright 2024 The Kubermatic Kubernetes Platform contributors.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
---
# Source: argocd-apps/templates/argocd-apps-kkp-core.yaml
apiVersion: argoproj.io/v1alpha1
kind: AppProject
metadata:
  name: kubermatic-config
  namespace: argocd
  finalizers:
    - resources-finalizer.argocd.argoproj.io
spec:
  description: ArgoCD Project for all core KKP configurations
  clusterResourceWhitelist:
  - group: '*'
    kind: '*'
  sourceRepos:
  - '*'
  destinations:
  - namespace: kubermatic
    server: https://kubernetes.default.svc
  - namespace: mla
    server: https://kubernetes.default.svc
  - namespace: iap
    server: https://kubernetes.default.svc
  - namespace: argocd
    server: https://kubernetes.default.svc
---
# Source: argocd-apps/templates/argocd-apps-usermla.yaml
apiVersion: argoproj.io/v1alpha1
kind: AppProject
metadata:
  name: mla
  namespace: argocd
  finalizers:
    - resources-finalizer.argocd.argoproj.io
spec:
  description: ArgoCD Project for all user cluster mla apps
  clusterResourceWhitelist:
  - group: '*'
    kind: '*'
  sourceRepos:
  - '*'
  destinations:
  - namespace: mla
    server: https://kubernetes.default.svc
---
# Source: argocd-apps/templates/argocd-apps-backup.yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: velero
  namespace: argocd
spec:
  project: default
  sources:
  - repoURL: 'https://github.com/kubermatic/kubermatic.git'
    path: ./charts/backup/velero
    targetRevision: notSet
    helm:
      valueFiles:
        - values.yaml
        - $values/dev/values.yaml
        - $values/dev/demo-master/values.yaml
  # This is your private repo where you have stored the values.yaml customizations which you used to provide to KKP installer. 
  - repoURL: 'https://github.com/dharapvj/kkp-using-argocd.git'
    targetRevision: dev-kkp-notSet
    path: .
    ref: values
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: velero
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
    automated: {}
---
# Source: argocd-apps/templates/argocd-apps-kkp-core.yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: oauth
  namespace: argocd
spec:
  project: default
  sources:
  - repoURL: 'https://github.com/kubermatic/kubermatic.git'
    path: ./charts/oauth
    targetRevision: notSet
    helm:
      valueFiles:
        - values.yaml
        - $values/dev/values.yaml
        - $values/dev/demo-master/values.yaml
  # This is your private repo where you have stored the values.yaml customizations which you used to provide to KKP installer. 
  - repoURL: 'https://github.com/dharapvj/kkp-using-argocd.git'
    targetRevision: dev-kkp-notSet
    path: .
    ref: values
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: oauth
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
    automated: {}
---
# Source: argocd-apps/templates/argocd-apps-kkp-core.yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: nginx-ingress-controller
  namespace: argocd
spec:
  project: default
  sources:
  - repoURL: 'https://github.com/kubermatic/kubermatic.git'
    path: ./charts/nginx-ingress-controller
    targetRevision: notSet
    helm:
      valueFiles:
        - values.yaml
        - $values/dev/values.yaml
        - $values/dev/demo-master/values.yaml
  # This is your private repo where you have stored the values.yaml customizations which you used to provide to KKP installer. 
  - repoURL: 'https://github.com/dharapvj/kkp-using-argocd.git'
    targetRevision: dev-kkp-notSet
    path: .
    ref: values
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: nginx-ingress-controller
  # nginx ingress app to be synced automatically so that we can access ArgoCD via it!
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
      - Retry=true
    retry:
      limit: 5
      backoff:
        duration: 5s
        factor: 2
        maxDuration: 3m
    automated: {}
---
# Source: argocd-apps/templates/argocd-apps-kkp-core.yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: cert-manager
  namespace: argocd
spec:
  project: default
  sources:
  - repoURL: 'https://github.com/kubermatic/kubermatic.git'
    path: ./charts/cert-manager
    targetRevision: notSet
    helm:
      parameters:
      - name: "cert-manager.installCRDs"
        value: "true"
      valueFiles:
        - values.yaml
        - $values/dev/values.yaml
        - $values/dev/demo-master/values.yaml
  # This is your private repo where you have stored the values.yaml customizations which you used to provide to KKP installer. 
  - repoURL: 'https://github.com/dharapvj/kkp-using-argocd.git'
    targetRevision: dev-kkp-notSet
    path: .
    ref: values
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: cert-manager
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
  # cert-manager app to be synced automatically so that we get certificates issued for KKP dashboard and ArgoCD to start with
    automated: {}
---
# Source: argocd-apps/templates/argocd-apps-kkp-core.yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: master-settings-demo-master
  namespace: argocd
spec:
  project: kubermatic-config
  sources:
  - repoURL: 'https://github.com/dharapvj/kkp-using-argocd.git'
    targetRevision: dev-kkp-notSet
    path: dev/settings
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: kubermatic
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
    automated: {}
---
# Source: argocd-apps/templates/argocd-apps-kkp-core.yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: seed-extras-demo-master
  namespace: argocd
spec:
  project: kubermatic-config
  sources:
  - repoURL: 'https://github.com/dharapvj/kkp-using-argocd.git'
    targetRevision: dev-kkp-notSet
    path: 'dev'
    directory:
      recurse: true
      exclude: '{**/values.yaml,**/values-usermla.yaml,**/k8cConfig.yaml}'
      include: '{common/*,demo-master/*}'
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: kubermatic
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
    # sometimes things take time and hence we should control the retry params
    retry:
      limit: 100
      backoff:
        duration: "20s"
        factor: 1
        maxDuration: "3m0s"
    automated: {}
---
# Source: argocd-apps/templates/argocd-apps-logging.yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: promtail
  namespace: argocd
spec:
  project: default
  # TODO: This assumes that we pull the charts from git repo. Currently, no support to pull charts from helm registry directly.
  sources:
  - repoURL: 'https://github.com/kubermatic/kubermatic.git'
    path: ./charts/logging/promtail
    targetRevision: notSet
    helm:
      valueFiles:
        - values.yaml
        - $values/dev/values.yaml
        - $values/dev/demo-master/values.yaml
  # This is your private repo where you have stored the values.yaml customizations which you used to provide to KKP installer. 
  - repoURL: 'https://github.com/dharapvj/kkp-using-argocd.git'
    targetRevision: dev-kkp-notSet
    path: .
    ref: values
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: logging
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
    automated: {}
---
# Source: argocd-apps/templates/argocd-apps-logging.yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: loki
  namespace: argocd
spec:
  project: default
  sources:
  - repoURL: 'https://github.com/kubermatic/kubermatic.git'
    path: ./charts/logging/loki
    targetRevision: notSet
    helm:
      valueFiles:
        - values.yaml
        - $values/dev/values.yaml
        - $values/dev/demo-master/values.yaml
  # This is your private repo where you have stored the values.yaml customizations which you used to provide to KKP installer. 
  - repoURL: 'https://github.com/dharapvj/kkp-using-argocd.git'
    targetRevision: dev-kkp-notSet
    path: .
    ref: values
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: logging
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
    automated: {}
---
# Source: argocd-apps/templates/argocd-apps-monitoring.yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: prometheus
  namespace: argocd
spec:
  project: default
  sources:
  - repoURL: 'https://github.com/kubermatic/kubermatic.git'
    path: ./charts/monitoring/prometheus
    targetRevision: notSet
    helm:
      valueFiles:
        - values.yaml
        - $values/dev/values.yaml
        - $values/dev/demo-master/values.yaml
  # This is your private repo where you have stored the values.yaml customizations which you used to provide to KKP installer. 
  - repoURL: 'https://github.com/dharapvj/kkp-using-argocd.git'
    targetRevision: dev-kkp-notSet
    path: .
    ref: values
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: monitoring
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
    automated: {}
---
# Source: argocd-apps/templates/argocd-apps-monitoring.yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: alertmanager
  namespace: argocd
spec:
  project: default
  sources:
  - repoURL: 'https://github.com/kubermatic/kubermatic.git'
    path: ./charts/monitoring/alertmanager
    targetRevision: notSet
    helm:
      valueFiles:
        - values.yaml
        - $values/dev/values.yaml
        - $values/dev/demo-master/values.yaml
  # This is your private repo where you have stored the values.yaml customizations which you used to provide to KKP installer. 
  - repoURL: 'https://github.com/dharapvj/kkp-using-argocd.git'
    targetRevision: dev-kkp-notSet
    path: .
    ref: values
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: monitoring
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
    automated: {}
---
# Source: argocd-apps/templates/argocd-apps-monitoring.yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: grafana
  namespace: argocd
spec:
  project: default
  sources:
  - repoURL: 'https://github.com/kubermatic/kubermatic.git'
    path: ./charts/monitoring/grafana
    targetRevision: notSet
    helm:
      valueFiles:
        - values.yaml
        - $values/dev/values.yaml
        - $values/dev/demo-master/values.yaml
  # This is your private repo where you have stored the values.yaml customizations which you used to provide to KKP installer. 
  - repoURL: 'https://github.com/dharapvj/kkp-using-argocd.git'
    targetRevision: dev-kkp-notSet
    path: .
    ref: values
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: monitoring
  # need to work around long annotations for dashboard configmaps
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
      - ServerSideApply=true
    automated: {}
---
# Source: argocd-apps/templates/argocd-apps-monitoring.yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: kube-state-metrics
  namespace: argocd
spec:
  project: default
  sources:
  - repoURL: 'https://github.com/kubermatic/kubermatic.git'
    path: ./charts/monitoring/kube-state-metrics
    targetRevision: notSet
    helm:
      valueFiles:
        - values.yaml
        - $values/dev/values.yaml
        - $values/dev/demo-master/values.yaml
  # This is your private repo where you have stored the values.yaml customizations which you used to provide to KKP installer. 
  - repoURL: 'https://github.com/dharapvj/kkp-using-argocd.git'
    targetRevision: dev-kkp-notSet
    path: .
    ref: values
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: monitoring
  # since kube-state-metrics comes with its own autoscaling resizer.. ignore resource value changes.
  ignoreDifferences:
  - group: apps
    kind: Deployment
    jsonPointers:
    - /spec/template/spec/containers/0/resources
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
    automated: {}
---
# Source: argocd-apps/templates/argocd-apps-monitoring.yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: node-exporter
  namespace: argocd
spec:
  project: default
  sources:
  - repoURL: 'https://github.com/kubermatic/kubermatic.git'
    path: ./charts/monitoring/node-exporter
    targetRevision: notSet
    helm:
      valueFiles:
        - values.yaml
        - $values/dev/values.yaml
        - $values/dev/demo-master/values.yaml
  # This is your private repo where you have stored the values.yaml customizations which you used to provide to KKP installer. 
  - repoURL: 'https://github.com/dharapvj/kkp-using-argocd.git'
    targetRevision: dev-kkp-notSet
    path: .
    ref: values
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: monitoring
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
    automated: {}
---
# Source: argocd-apps/templates/argocd-apps-monitoring.yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: blackbox-exporter
  namespace: argocd
spec:
  project: default
  sources:
  - repoURL: 'https://github.com/kubermatic/kubermatic.git'
    path: ./charts/monitoring/blackbox-exporter
    targetRevision: notSet
    helm:
      valueFiles:
        - values.yaml
        - $values/dev/values.yaml
        - $values/dev/demo-master/values.yaml
  # This is your private repo where you have stored the values.yaml customizations which you used to provide to KKP installer. 
  - repoURL: 'https://github.com/dharapvj/kkp-using-argocd.git'
    targetRevision: dev-kkp-notSet
    path: .
    ref: values
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: monitoring
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
    automated: {}
---
# Source: argocd-apps/templates/argocd-apps-monitoring.yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: seed-mla-iap
  namespace: argocd
spec:
  project: default
  sources:
  - repoURL: 'https://github.com/kubermatic/kubermatic.git'
    path: ./charts/iap
    targetRevision: notSet
    helm:
      valueFiles:
        - values.yaml
        - $values/dev/values.yaml
        - $values/dev/demo-master/values.yaml
  # This is your private repo where you have stored the values.yaml customizations which you used to provide to KKP installer. 
  - repoURL: 'https://github.com/dharapvj/kkp-using-argocd.git'
    targetRevision: dev-kkp-notSet
    path: .
    ref: values
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: iap
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
    automated: {}
---
# Source: argocd-apps/templates/argocd-apps-storage.yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: minio
  namespace: argocd
spec:
  project: default
  sources:
  - repoURL: 'https://github.com/kubermatic/kubermatic.git'
    path: ./charts/minio
    targetRevision: notSet
    helm:
      valueFiles:
        - values.yaml
        - $values/dev/values.yaml
        - $values/dev/demo-master/values.yaml
  # This is your private repo where you have stored the values.yaml customizations which you used to provide to KKP installer. 
  - repoURL: 'https://github.com/dharapvj/kkp-using-argocd.git'
    targetRevision: dev-kkp-notSet
    path: .
    ref: values
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: minio
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
    automated: {}
---
# Source: argocd-apps/templates/argocd-apps-usermla.yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: user-mla-minio
  namespace: argocd
spec:
  project: mla
  sources:
  - repoURL: 'https://github.com/kubermatic/kubermatic.git'
    path: ./charts/mla/minio
    targetRevision: notSet
    helm:
      releaseName: minio
      valueFiles:
        - values.yaml
        - $values/dev/demo-master/values-usermla.yaml
  # This is your private repo where you have stored the values.yaml customizations which you used to provide to KKP installer. 
  - repoURL: 'https://github.com/dharapvj/kkp-using-argocd.git'
    targetRevision: dev-kkp-notSet
    path: .
    ref: values
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: mla
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
    automated: {}
---
# Source: argocd-apps/templates/argocd-apps-usermla.yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: user-mla-secrets
  namespace: argocd
spec:
  project: mla
  sources:
  - repoURL: 'https://github.com/kubermatic/kubermatic.git'
    path: ./charts/mla/mla-secrets
    targetRevision: notSet
    helm:
      valueFiles:
        - values.yaml
        - $values/dev/demo-master/values-usermla.yaml
  # This is your private repo where you have stored the values.yaml customizations which you used to provide to KKP installer. 
  - repoURL: 'https://github.com/dharapvj/kkp-using-argocd.git'
    targetRevision: dev-kkp-notSet
    path: .
    ref: values
    helm:
      # TODO: we should allow users to override releaseName. This will help in backward compatibility to existing helm charts AND not create duplicate resources
      # releaseName: mla-secrets
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: mla
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
    automated: {}
---
# Source: argocd-apps/templates/argocd-apps-usermla.yaml
---
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: user-mla-minio-lifecycle-manager
  namespace: argocd
spec:
  project: mla
  sources:
  - repoURL: 'https://github.com/kubermatic/kubermatic.git'
    path: ./charts/mla/minio-lifecycle-mgr
    targetRevision: notSet
    helm:
      valueFiles:
        - values.yaml
        - $values/dev/demo-master/values-usermla.yaml
  # This is your private repo where you have stored the values.yaml customizations which you used to provide to KKP installer. 
  - repoURL: 'https://github.com/dharapvj/kkp-using-argocd.git'
    targetRevision: dev-kkp-notSet
    path: .
    ref: values
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: mla
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
    automated: {}
---
# Source: argocd-apps/templates/argocd-apps-usermla.yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: user-mla-grafana
  namespace: argocd
spec:
  project: mla
  sources:
  - repoURL: 'https://github.com/kubermatic/kubermatic.git'
    path: ./charts/mla/grafana
    targetRevision: notSet
    helm:
      releaseName: grafana
      valueFiles:
        - values.yaml
        - $values/dev/demo-master/values-usermla.yaml
  # This is your private repo where you have stored the values.yaml customizations which you used to provide to KKP installer. 
  - repoURL: 'https://github.com/dharapvj/kkp-using-argocd.git'
    targetRevision: dev-kkp-notSet
    path: .
    ref: values
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: mla
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
    automated: {}
---
# Source: argocd-apps/templates/argocd-apps-usermla.yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: user-mla-consul
  namespace: argocd
spec:
  project: mla
  sources:
  - repoURL: 'https://github.com/kubermatic/kubermatic.git'
    path: ./charts/mla/consul
    targetRevision: notSet
    helm:
      releaseName: consul
      valueFiles:
        - values.yaml
        - $values/dev/demo-master/values-usermla.yaml
  # This is your private repo where you have stored the values.yaml customizations which you used to provide to KKP installer. 
  - repoURL: 'https://github.com/dharapvj/kkp-using-argocd.git'
    targetRevision: dev-kkp-notSet
    path: .
    ref: values
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: mla
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
    automated: {}
---
# Source: argocd-apps/templates/argocd-apps-usermla.yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: user-mla-alertmanager-proxy
  namespace: argocd
spec:
  project: mla
  sources:
  - repoURL: 'https://github.com/kubermatic/kubermatic.git'
    path: ./charts/mla/alertmanager-proxy
    targetRevision: notSet
    helm:
      valueFiles:
        - values.yaml
        - $values/dev/demo-master/values-usermla.yaml
  # This is your private repo where you have stored the values.yaml customizations which you used to provide to KKP installer. 
  - repoURL: 'https://github.com/dharapvj/kkp-using-argocd.git'
    targetRevision: dev-kkp-notSet
    path: .
    ref: values
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: mla
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
    automated: {}
---
# Source: argocd-apps/templates/argocd-apps-usermla.yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: user-mla-cortex
  namespace: argocd
spec:
  project: mla
  sources:
  - repoURL: 'https://github.com/kubermatic/kubermatic.git'
    path: ./charts/mla/cortex
    targetRevision: notSet
    helm:
      releaseName: cortex
      valueFiles:
        - values.yaml
        - $values/dev/demo-master/values-usermla.yaml
  # This is your private repo where you have stored the values.yaml customizations which you used to provide to KKP installer. 
  - repoURL: 'https://github.com/dharapvj/kkp-using-argocd.git'
    targetRevision: dev-kkp-notSet
    path: .
    ref: values
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: mla
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
    automated: {}
---
# Source: argocd-apps/templates/argocd-apps-usermla.yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: user-mla-loki-distributed
  namespace: argocd
spec:
  project: mla
  sources:
  - repoURL: 'https://github.com/kubermatic/kubermatic.git'
    path: ./charts/mla/loki-distributed
    targetRevision: notSet
    helm:
      valueFiles:
        - values.yaml
        - $values/dev/demo-master/values-usermla.yaml
  # This is your private repo where you have stored the values.yaml customizations which you used to provide to KKP installer. 
  - repoURL: 'https://github.com/dharapvj/kkp-using-argocd.git'
    targetRevision: dev-kkp-notSet
    path: .
    ref: values
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: mla
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
    automated: {}
---
# Source: argocd-apps/templates/argocd-apps-usermla.yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: user-mla-iap
  namespace: argocd
spec:
  project: mla
  sources:
  - repoURL: 'https://github.com/kubermatic/kubermatic.git'
    path: ./charts/iap
    targetRevision: notSet
    helm:
      valueFiles:
        - values.yaml
        - $values/dev/demo-master/values-usermla.yaml
  # This is your private repo where you have stored the values.yaml customizations which you used to provide to KKP installer. 
  - repoURL: 'https://github.com/dharapvj/kkp-using-argocd.git'
    targetRevision: dev-kkp-notSet
    path: .
    ref: values
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: mla
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
    automated: {}
